corehq/apps/locations/tests/test_api_resources.py

```diff
@@ -409,3 +414,55 @@ def test_patch_list_missing_location_id(self):
         self.assertEqual(response.json(),
                          {'error': "Could not update: could not find location with"
                                    f" given ID {unknown_location_id} on the domain."})
+
+
+class TestV0_6LocationsList(APIResourceTest):
+    api_name = 'v0.6'
+    resource = v0_6.LocationResource
+    location_type_names = ['state', 'county', 'city']
+    location_structure = [
+        ('Massachusetts', [
+            ('Middlesex', [
+                ('Cambridge', []),
+                ('Somerville', []),
+            ]),
+            ('Suffolk', [
+                ('Boston', []),
+            ])
+        ])
+    ]
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        with freeze_time(datetime(2025, 1, 1)):
+            cls.location_types, cls.locations = setup_locations_and_types(
+                cls.domain.name,
+                cls.location_type_names,
+                [],
+                cls.location_structure,
+            )
+            cls.locations['Middlesex'].location_id = 'middlesex_uuid'
+            cls.locations['Middlesex'].save()
+
+        with freeze_time(datetime(2025, 4, 1)):
+            cls.locations['Boston'].save()
+            cls.locations['Cambridge'].save()
+
+    @generate_cases([
+        ('site_code=boston', ['boston']),
+        ('last_modified.gte=2025-03-31', ['boston', 'cambridge']),
+        ('last_modified.gte=2020-01-01&last_modified.lt=2025-03-31', [
+            'massachusetts', 'middlesex', 'somerville', 'suffolk',
+        ]),
+        ('last_modified.gt=2025-03-31&last_modified.lte=2025-03-30', []),
+        ('location_type_code=county', ['middlesex', 'suffolk']),
+        ('parent_location_id=middlesex_uuid', ['cambridge', 'somerville']),
+    ])
+    def test_api_filters(self, querystring, expected):
+        url = f"{self.list_endpoint}?{querystring}"
+        response = self._assert_auth_get_resource(url)
+        self.assertEqual(response.status_code, 200)
+        self.assertItemsEqual([
+            loc['site_code'] for loc in response.json()['objects']
```

Comment:
Because the only thing the tests specify is the location name, I think this might be clearer if we changed `site_code` to `name`. I wasn't clear how `Boston` became `boston`, and had to do some digging to find the line in the `save` code. Because one of these test cases filters on the site-code, it would be nice to explicitly assign Boston's site code.

Reply:
Updated in https://github.com/dimagi/commcare-hq/pull/36285/commits/1b9f469a70f24204abedbc8f970497f6d76ed1e6 (and also let you filter by `name`, because why not)